fplll/pruner.h

```diff
@@ -0,0 +1,568 @@
+/* Copyright (C) 2015-2016 Martin Albrecht, Leo Ducas.
+
+   This file is part of fplll. fplll is free software: you
+   can redistribute it and/or modify it under the terms of the GNU Lesser
+   General Public License as published by the Free Software Foundation,
+   either version 2.1 of the License, or (at your option) any later version.
+
+   fplll is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTAbILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+   GNU Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public License
+   along with fplll. If not, see <http://www.gnu.org/licenses/>. */
+
+
+#include <array>
+#include "factorial.const"
+#include "ballvol.const"
+#include "fplll.h"
+
+FPLLL_BEGIN_NAMESPACE
+
+// This file propose an implementation of the numerically optimized
+// pruning strategy of [GNR10]. 
+
+// Many details for implemantation follows from the thesis [Chen13]
+// Some simplification have been made, for example we restrict ourselves
+// to ``even vector bounds'' i.e., the bound at indices 2i and 2i+1 are
+// kept equal, as to allow volume estimation by pure integration 
+// (see [GNR10,Chen13])
+
+// naming conventions:
+// b is for bound (squared)
+// pv is for partial volumes (NOT squared)
+// r is for gram schmidt length (squared). Automatically renormalized
+// to avoid overflowing partial volumes
+// p is for polynomial
+
+// inside this code, b,pv,and R are in reversed order
+// as to conform with the algorithm desciption of [Chen13]
+// reversing output and ouput is done by the C extern function
```

Comment:
'extern "C"' has a specific meaning in C++ (link using C convention, not C++ convention). This is not what you mean here, right?